demisto_sdk/common/rest_api.py

```diff
@@ -0,0 +1,192 @@
+import os
+import json
+from urllib.parse import urljoin
+
+import requests
+
+from demisto_sdk.common.tools import print_v
+
+
+DEMISTO_API_KEY_ENV = 'DEMISTO_API_KEY'
+INTEGRATION_UPLOAD_PATH = "/settings/integration-conf/upload"
+ENTRY_EXECUTE_PATH = "/entry/execute/sync"
+ENTRY_DOWNLOAD_PATH = "/entry/download/"
+INVESTIGATION_SEARCH_PATH = "/investigations/search"
+
+
+class Client:
+    """Demisto REST API Client
+    
+    Args:
+        base_url (str): URL of Demisto instance
+        verify_cert (bool, optional): Verify remote certificate. Defaults to True.
+        verbose (bool, optional): Verbose logging.
+    """
+    def __init__(self, base_url: str, verify_cert: bool = True, verbose: bool = False):
+        self.base_url = base_url
+        self.verify_cert = verify_cert
+        self.log_verbose = verbose
+
+    def get_api_key(self):
+        """Retrieve the API Key
+        
+        Raises:
+            RuntimeError: if the API Key environment variable is not found
+        
+        Returns:
+            str: API Key
+        """
+        ans = os.environ.get(DEMISTO_API_KEY_ENV, None)
+        if ans is None:
+            raise RuntimeError(f'Error: Environment variable {DEMISTO_API_KEY_ENV} not found')
+
+        return ans
+
+    def upload_integration(self, ymlf):
+        """Upload an integration YML file to a remote Demisto instance.
+
+        Args:
+            ymlf (stream): The open file object to be uploaded
+        """
+        api_key = self.get_api_key()
+
+        api_url = urljoin(self.base_url, INTEGRATION_UPLOAD_PATH)
+        files = {'file': ymlf}
+
+        print_v(f'Uploading integration to {api_url}', self.log_verbose)
```

Comment:
i think that there is no need to specify the whole path, just the instance name is relevant here
```suggestion
        print_v(f'Uploading integration to {self.base_url}', self.log_verbose)
```